openthread: fix logging configuration

maciejbaczmanski opened the pull request:
Currently, `CONFIG_OPENTHREAD_LOG_LEVEL` is used to set log level in OT and register log modules in Zephyr. OpenThread allows 5 levels, causing issues when `OPENTHREAD_LOG_LEVEL_DEBG` is selected ad we are trying to register modules with unknown lvl.

This commit adds `CONFIG_OPENTHREAD_MODULE_LOG_LEVEL` which is aligned for Zephyr's log levels.

maciejbaczmanski committed:
openthread: fix logging configuration

Currently, `CONFIG_OPENTHREAD_LOG_LEVEL` is used to set log level
in OT and register log modules in Zephyr. OpenThread allows 5
levels, causing issues when `OPENTHREAD_LOG_LEVEL_DEBG` is
selected ad we are trying to register modules with unknown lvl.

This commit adds `CONFIG_OPENTHREAD_MODULE_LOG_LEVEL` which is
aligned for Zephyr's log levels.

Signed-off-by: Maciej Baczmanski <maciej.baczmanski@nordicsemi.no>

kkasperczyk-no reviewed (approved):
I think it would be nice to add help explaining that OPENTHREAD_LOG_LEVEL applies to OpenThread stack and OPENTHREAD_MODULE_LOG_LEVEL applies to zephyr platform (or rename it to OPENTHREAD_PLATFORM)

maciejbaczmanski commented:
> I think it would be nice to add help explaining that OPENTHREAD_LOG_LEVEL applies to OpenThread stack and OPENTHREAD_MODULE_LOG_LEVEL applies to zephyr platform (or rename it to OPENTHREAD_PLATFORM)

good idea, renamed and added helps. thanks
